File: Compose-Bookshelf-App/app/src/test/java/com/example/bookshelfapp/fake/ThumbnailFakeDataSource.kt
package com.example.bookshelfapp.fake

import com.example.bookshelfapp.data.BookThumbnailDataSource
import com.example.bookshelfapp.model.GoogleBookRetrieveApiModel
import com.example.bookshelfapp.model.RetrieveVolumeInfoModel
import com.example.bookshelfapp.model.ThumbnailModel

class ThumbnailFakeDataSource : BookThumbnailDataSource {
    override suspend fun getThumbnails(bookId: String): GoogleBookRetrieveApiModel {
        return GoogleBookRetrieveApiModel(
            volumeInfo = RetrieveVolumeInfoModel(
                imageLinks = ThumbnailModel(
                    thumbnail = "thumbnail"
                )
            )
        )
    }
}

File: SwiftUI-Bookshelf-App/Bookshelf appTests/Fake/ThumbnailFakeDataSource.swift
//
//  ThumbnailFakeDataSource.swift
//  Bookshelf appTests
//
//  Created by Jaehwa Noh on 1/29/24.
//

@testable import Bookshelf_app

struct ThumbnailFakeDataSource: BookThumbnailDataSource {
    func getThumbnails(bookId: String) async throws -> Bookshelf_app.GoogleBookRetrieveApiModel {
        return GoogleBookRetrieveApiModel(
            volumeInfo: RetrieveVolumeInfoModel(
                imageLinks: ThumbnailModel(
                    thumbnail: "thumbnail")
            )
        )
    }
}
